lib/network_config.ml

```diff
@@ -64,13 +65,27 @@ let read_management_conf () =
 		in
 		let phy_interface = {default_interface with persistent_i = true} in
 		let bridge_interface = {default_interface with ipv4_conf; ipv4_gateway; persistent_i = true} in
-		let bridge = {default_bridge with
-			bridge_mac = Some mac;
-			ports = [device, {default_port with interfaces = [device]}];
-			persistent_b = true
-		} in
-		{interface_config = [device, phy_interface; bridge_name, bridge_interface];
-			bridge_config = [bridge_name, bridge];
+		let interface_config, bridge_config =
+			let primary_bridge_conf = {default_bridge with
+                        	bridge_mac = Some mac;
+	                        ports = [device, {default_port with interfaces = [device]}];
+        	                persistent_b = true
+                	} in
+			if vlan <> "" && not (String.startswith "xenbr" bridge_name) && bridge_name <> "" then begin
+				let parent = "xenbr" ^ (String.sub device 3 (String.length device - 3)) in
+				let secondary_bridge_conf = {default_bridge with
+					vlan = Some (parent, int_of_string vlan);
+					bridge_mac = (Some mac);
+					persistent_b = true
+				} in
+				[device, phy_interface; parent, phy_interface; bridge_name, bridge_interface],
```

Comment:
Since we are not assigning IP to parent bridge we should not use `ipv4_conf` `ipv4_gateway`.
We can change the variable names but doesn't make any functional difference.